torch/_inductor/lowering.py

```diff
@@ -2866,7 +2865,7 @@ def max_pool2d_with_indices(
     new_size = list(batch) + [h_out, w_out]
     window_size = kernel_size[0] * kernel_size[1]
 
-    if window_size > 25:
+    if window_size > 25 or (dilation != 1 and any(d != 1 for d in dilation)):
```

Comment:
I don't think this would be correct when `dilation` is an int but not 1.

Reply:
Thank you for your review! I thought this function might not have a chance to accept this kind of parameter, and I would like try to find some code evidence.

Reply:
Do you mean things like `dilation == 2` won't happen? Why?

Reply:
This depands on the user's usage method. In the most case like the following snippet.
```python
def fn(a):
    return torch.ops.aten.max_pool2d_with_indices(x, [3, 2], [2, 1], 1)
x = torch.randn([2, 2, 3, 6])
fn_compiled = torch.compile(fn)
y = fn_compiled(x)
```
It works fine, even though the padding parameter is a scalar in Python. The magic is probably related with `__torch_dispatch__`, a scalar is converted to a list in C++.
But if the user call this function directly, it will definitely fail. And I think the problem was preexisting, should I fix it in this pr? @jgong5

Reply:
I would suggest you add an assertion if you don't believe a condition would happen, or do the right conditional checks.

Reply:
updated